src/library/scala/collection/immutable/SortedMap.scala

```diff
@@ -108,11 +108,11 @@ trait SortedMapOps[K, +V, +CC[X, +Y] <: Map[X, Y] with SortedMapOps[X, Y, CC, _]
   override def updatedWith[V1 >: V](key: K)(remappingFunction: Option[V] => Option[V1]): CC[K, V1] = {
     // Implementation has been copied from `MapOps`
```

Comment:
anyone know why the implementation is copied?

Reply:
The commet above says

>   // We override these methods to fix their return type (which would be `Map` otherwise)

In `SortedMapOps[K, +V, +CC[X, Y] <: ..., +C <: ...] extends MapOps[K, V, Map, C]`, note how `CC` is not forwarded to `MapOps`.